plugin/src/main/java/org/opensearch/ml/utils/MLNodeUtils.java

```diff
@@ -109,10 +109,14 @@ public static String processRemoteInferenceInputDataSetParametersValue(String in
                 if (value.isTextual()) {
                     String textValue = value.asText();
                     try {
-                        // Try to parse the string as JSON
-                        JsonNode parsedValue = mapper.readTree(textValue);
-                        // If successful, replace the string with the parsed JSON
-                        parametersNode.set(key, parsedValue);
+                        // Only try to parse as JSON if it looks like a JSON object or array
+                        if ((textValue.startsWith("{") && textValue.endsWith("}"))
+                            || (textValue.startsWith("[") && textValue.endsWith("]"))) {
+                            // Try to parse the string as JSON
+                            JsonNode parsedValue = mapper.readTree(textValue);
```

Comment:
this method is going to impact boarder use cases, not only the integer within the string